modules/packages/MPI.chpl

```diff
@@ -260,6 +265,20 @@ module MPI {
      processes (world size)
      */
   proc initialize() {
+    // If we are running using the uGNI layer, then the following hack
+    // appears to be necessary in order to run MPI, as well as Chapel
+    // See : https://hpcrdm.lbl.gov/pipermail/upc-users/2014-May/002061.html
+    if isUGNI {
+      coforall loc in Locales do on loc {
+          // This must be done on all locales!
+          var pmiGniCookie = C_Env.getenv("PMI_GNI_COOKIE") : string;
+          if !pmiGniCookie.isEmptyString() {
+            C_Env.setenv("PMI_GNI_COOKIE",("%i".format(pmiGniCookie:int+1)).c_str(),1);
+          }
```

Comment:
@gbtitus (our resident ugni expert) do you have any thoughts on this?

Curiously is looks like gasnet doesn't actually do this automatically: https://github.com/chapel-lang/chapel/blob/master/third-party/gasnet/gasnet-src/gemini-conduit/gasnet_core.c#L117-L143